src/usd/usd_parser/utils.cc

```diff
@@ -0,0 +1,666 @@
+              else
+              {
+                pxr::UsdShadeInput reflectionRoughnessConstantShaderInput =
+                  variantshader.GetInput(pxr::TfToken("reflection_roughness_constant"));
+                float reflectionRoughnessConstant;
+                reflectionRoughnessConstantShaderInput.Get(&reflectionRoughnessConstant);
+                pbrWorkflow.SetRoughness(reflectionRoughnessConstant);
+                isPBR = true;
+              }
+            }
+            else if (input.GetBaseName() == "enable_emission")
+            {
+              pxr::UsdShadeInput enableEmissiveShaderInput =
+                variantshader.GetInput(pxr::TfToken("enable_emission"));
+              enableEmissiveShaderInput.Get(&enableEmission);
+              std::cerr << "\tenableEmission " << enableEmission << '\n';
+            }
+            else if (input.GetBaseName() == "emissive_color")
+            {
+                pxr::UsdShadeInput emissiveColorShaderInput =
+                  variantshader.GetInput(pxr::TfToken("emissive_color"));
+                if (emissiveColorShaderInput.Get(&emissiveColor))
+                {
+                  std::cerr << "\temissiveColor " << emissiveColor << '\n';
+
+                  emissiveColorCommon = ignition::math::Color(
+                    emissiveColor[0],
+                    emissiveColor[1],
+                    emissiveColor[2]);
+                }
+            }
+          }
+
+          if (enableEmission)
+          {
+            material.SetEmissive(emissiveColorCommon);
+          }
+
+          if (isPBR)
+          {
+            sdf::Pbr pbr;
+            pbr.SetWorkflow(sdf::PbrWorkflowType::METAL, pbrWorkflow);
+            material.SetPbrMaterial(pbr);
+          }
+        }
+      }
+    }
+    return material;
+  }
+
+  void GetAllTransforms(
+    const pxr::UsdPrim &_prim,
+    USDData &_usdData,
+    std::vector<ignition::math::Pose3d> &_tfs,
+    ignition::math::Vector3d &_scale,
+    const std::string &_name)
+  {
+    pxr::UsdPrim parent = _prim;
+    double metersPerUnit = 1.0;
+    while(parent)
+    {
+      if (pxr::TfStringify(parent.GetPath()) == _name)
+      {
+        return;
+      }
+
+      Transforms t = ParseTransform(parent, _usdData);
+
+      ignition::math::Pose3d pose;
+      _scale *= t.scale;
+
+      std::cout << "parent.GetPath().GetName() " << parent.GetPath().GetName() << '\n';
+
+      std::pair<std::string, std::shared_ptr<USDStage>> data =
+        _usdData.findStage(parent.GetPath().GetName());
+
+      if (data.second != nullptr)
+      {
+        metersPerUnit = data.second->_metersPerUnit;
+        std::string upAxis = data.second->_upAxis;
+        std::cerr << "upAxis " << upAxis << '\n';
+      //   if(upAxis == "Y")
+      //   {
+      //     ignition::math::Matrix4d m(
+      //       1, 0, 0, 0,
+      //       0, 0, -1, 0,
+      //       0, 1, 0, 0,
+      //       0, 0, 0, 1);
+      //     t.translate = m * t.translate;
```

Comment:
For the scenario of `upAxis == "Y"`, is there a reason why the rotation matrix is 4x4? It should be 3x3, since we are in a 3d space. Reading the [USD upAxis docs](https://graphics.pixar.com/usd/release/api/group___usd_geom_up_axis__group.html#details), I found the following information that might be useful:
```
UsdGeom stipulates a right-handed coordinate system.
Therefore, when viewing a UsdStage with a "Y" up axis, the stage's Z axis will be pointing out of the screen, and when viewing a UsdStage with a "Z" up axis, the stage's Y axis will be pointing into the screen.
```

So, if we are looking at the screen, this means that X will always be pointing to the right (i.e., the X coordinate will never change regardless of the `upAxis`). So, when the `upAxis` is `Y`, I think we need to perform a 90 degree rotation in the YZ plane, from the Y axis to the Z axis (for a visual representation of this, look at `Ryz` [here](https://robotics.stackexchange.com/a/19616)). So, I think that the rotation matrix (`m`) should be this (it worked for a few points I tested on paper/by hand):
```
1     0       0              1 0 0
0  cos(90)  -sin(90)    =    0 0 -1
0  sin(90)   cos(90)         0 1  0
```
Note that this is the matrix `m` you currently have, but with the last row/column removed.

Reply:
It sounds like this is now resolved by https://github.com/ignitionrobotics/sdformat/pull/825